src/Doctrine/EntityRelation.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Doctrine;
+
+/**
+ * @internal
+ */
+final class EntityRelation
+{
+    const MANY_TO_ONE = 'ManyToOne';
+    const ONE_TO_MANY = 'OneToMany';
+    const MANY_TO_MANY = 'ManyToMany';
+    const ONE_TO_ONE = 'OneToOne';
+
+    private $type;
+
+    private $owningClass;
+
+    private $inverseClass;
+
+    private $owningProperty;
+
+    private $inverseProperty;
+
+    private $isNullable = false;
+
+    private $orphanRemoval = false;
+
+    private $mapInverseRelation = true;
```

Comment:
This is one of those tough spots. What are the actual *downsides* of mapping the inverse side? The generator actually generates setter/adder/remover methods on the inverse side that *set* the owning side, which makes the "I set the inverse side and nothing happened" problem lessened or completely eliminated.

Reply:
Inverse 1:1 assocations could be a performance nightmare. Also synchronization is.

Reply:
I'm going to add a question to ask the user if they want an inverse side, instead of always adding it. Then, it's their fault ;). Plus, we only synchronize the owning side (from the inverse side) to minimize performance. If you're not going to set the inverse side, then you won't call these methods and have the penalty anyways. But if you *do* set the inverse side, then it's *really* nice to synchronize and I think the penalty is worth it.